pkg/proxy/policy/policy.go

```diff
@@ -10,10 +11,60 @@ import (
 	"github.com/elazarl/goproxy"
 )
 
+var ruleRegistry = map[string]func() Rule{}
+
+// RegisterRule adds a rule to the rule registry.
+// The rule registry contains implementations of the Rule interface recognized
+// by the proxy. Updates to the policy will try to unmarshal json structs into the
+// registered rules. Expects the rule name and a constructor returning the Rule type.
+func RegisterRule(rulename string, constructor func() Rule) {
```

Comment:
oh yep I didn't think that through. i guess the only way i can think of to avoid this would be to push the implementation cost into the Rule interface by requiring a "Name" or something for the string ID and a "Make" or "New" method (gross, I know) to provide the constructor. Other than that, looks like reflection would be required :/

all in all, I think the current design is fine